docs/components/emails.rst

```diff
@@ -1,13 +1,434 @@
+                    'labels' => ['label1', 'label2'],
+                    'data'   => [
+                        'label1' => [100,200],
+                        'label2' => [200,300]
+                    ],
+                    'step_width' => 10
+                ],
+                'supportTemplate' => 'HelloWorldBundle:SubscribedEvents\AbTest:bargraph.html.php'
+            ]);
+        }
+    }
+
+.. code-block:: PHP
+
+    <?php
+    // plugins/HelloWorldBundle/Views/SubscribedEvents/AbTest/bargraph.html.php
+
+    declare(strict_types=1);
+
+    $support = $results['support'];
+    $label   = 'My chart label';
+    $chart   = new \Mautic\CoreBundle\Helper\Chart\BarChart($support['labels']);
+
+    if ($support['data']) {
+        foreach ($support['data'] as $datasetLabel => $values) {
+            $chart->setDataset($datasetLabel, $values);
+        }
+    }
+    ?>
+
+    <div class="panel ovf-h bg-auto bg-light-xs abtest-bar-chart">
+        <div class="panel-body box-layout">
+            <div class="col-xs-8 va-m">
+                <h5 class="text-white dark-md fw-sb mb-xs">
+                    <?php echo $label; ?>
+                </h5>
+            </div>
+            <div class="col-xs-4 va-t text-right">
+                <h3 class="text-white dark-sm"><span class="fa fa-bar-chart"></span></h3>
+            </div>
+        </div>
+        <?php echo $view->render(
+            'MauticCoreBundle:Helper:chart.html.php',
+            ['chartData' => $chart->render(), 'chartType' => 'bar', 'chartHeight' => 300]
+        ); ?>
+    </div>
+
+.. vale off
+
+Monitored Inbox Integration
+---------------------------
+
+.. vale on
+
+Plugins have access to hook into the ``mautic:email:fetch`` command to fetch email from a specific inbox/folder and process the content of the message.
+The Plugin also has access to inject specific search criteria for the messages to be processed.
+
+To do this, the Plugin needs to add an event listener for three events:
+
+1. ``EmailEvents::MONITORED_EMAIL_CONFIG`` This event is dispatched to inject the fields into Mautic's Configuration to configure the IMAP inbox and folder that should be monitored.
+2. ``EmailEvents::EMAIL_PRE_FETCH`` This event is dispatched during the execution of the ``mautic:email:fetch`` command. It's used to inject search criteria for the messages desired.
+3. ``EmailEvents::EMAIL_PARSE`` This event parses the messages fetched by the command.
+
+.. code-block:: PHP
+
+    <?php
+    // plugins/HelloWorldBundle/EventListener/MonitoredInboxSubscriber.php
+
+    declare(strict_types=1);
+
+    namespace MauticPlugin\HelloWorldBundle\EventListener;
+
+    use Mautic\EmailBundle\EmailEvents;
+    use Mautic\EmailBundle\Event\MonitoredEmailEvent;
+    use Mautic\EmailBundle\Event\ParseEmailEvent;
+    use Mautic\EmailBundle\MonitoredEmail\Mailbox;
+    use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+    class MonitoredInboxSubscriber implements EventSubscriberInterface
```

Comment:
Replaced `CommonSubscriber` (removed) from the old docs with `EventSubscriberInterface`